lib/jsonschema.lua

```diff
@@ -479,8 +479,8 @@ local function typeexpr(ctx, jsontype, datatype, tablekind)
   elseif jsontype == 'table' then
     return sformat(' %s == "table" ', datatype)
   elseif jsontype == 'integer' then
-    return sformat(' (%s == "number" and %s(%s, 1.0) == 0.0) ',
-      datatype, ctx:libfunc('math.fmod'), ctx:param(1))
+    return sformat(' ((%s == "number" or (%s == "cdata" and tonumber(%s) ~= nil)) and %s %% 1.0 == 0.0) ',
```

Comment:
we need a test case to check it.

Reply:
here is an example: https://github.com/api7/jsonschema/commit/7ff3366cf53088edc2ffc8ef84bb9cee9a29c831#diff-38b895a33e0e96aa710cdabe2de3d1118ddcd94ea567781b1b9f0fb0062c2858R137

Reply:
I added a couple of tests